src/wzapi.cpp

```diff
@@ -1633,9 +1633,9 @@ static int get_first_available_component(int player, int capacity, const wzapi::
 	return -1; // no available component found in list
 }
 
-static DROID_TEMPLATE *makeTemplate(int player, const std::string &templName, const wzapi::string_or_string_list& _body, const wzapi::string_or_string_list& _propulsion, const wzapi::va_list<wzapi::string_or_string_list>& _turrets, int capacity, bool strict)
+static std::unique_ptr<DROID_TEMPLATE> makeTemplate(int player, const std::string &templName, const wzapi::string_or_string_list& _body, const wzapi::string_or_string_list& _propulsion, const wzapi::va_list<wzapi::string_or_string_list>& _turrets, int capacity, bool strict)
 {
-	DROID_TEMPLATE *psTemplate = new DROID_TEMPLATE;
+	std::unique_ptr<DROID_TEMPLATE> psTemplate = std::unique_ptr<DROID_TEMPLATE>(new DROID_TEMPLATE);
```

Comment:
We could initialize the `DROID_TEMPLATE` in the stack, and then at the end of the function initialize the `unique_ptr` using the copy constructor, to avoid allocating in the heap in case of error. But it seems like the error cases are really "exceptional" so it wouldn't make much of a difference, right?

Reply:
Correct - I don't believe failure cases are super common / frequent.